core-java-modules/core-java-lang-5/src/main/java/com/baeldung/generics/Cat.java

```diff
@@ -0,0 +1,32 @@
+package com.baeldung.generics;
+
+class Cat extends Animal implements Comparable<Cat> {
+    public Cat(String type, String name) {
+        super(type, name);
+    }
+
+    @Override
+    public String makeSound() {
+        return "Meow";
+    }
+
+    @Override
+    public int compareTo(Cat cat) {
+        return this.getName()
+            .length() - cat.getName()
+            .length();
```

Comment:
In this case, we can put together, in the same line the call to getName() and length(). In fact, it can be written in just one line and it would easier to read it.